ouroboros-consensus-diffusion/changelog.d/20230612_124926_armandoifsantos_dynamic_block_forging.md

```diff
@@ -0,0 +1,35 @@
+<!--
+A new scriv changelog fragment.
+
+Uncomment the section that is right (remove the HTML comment wrapper).
+-->
+
+<!--
+### Patch
+
+- A bullet item for the Patch category.
+
+-->
+
+### Non-Breaking
+
+- Change how diffusion initializes block forging credentials: now there's a `TMVar` that
+  monitors so we can enable/disable block forging dynamically.
+  - There's also a `blockForgingController` that keeps an eye at this `TMVar`.
+
+- Adds new trace `TraceAdoptionThreadDied SlotNo blk` to `TraceForgeEvent`
+
+### Breaking
+
+- Adds a new argument to `RekeyM` type, due to the extraction of the block forging
+  credentials over at `ouroboros-consensus`.
+  - Refactor to accommodate this change
+
+- Adds block forging credentials to `TestNodeInitialization`
+
+- Changes the type signature of most functions to receive the block forging credentials as
+  an argument, since it now doesn't come in the usual bundle.
+
+- Added `setBlockForging` to `NodeKernel` which must be used to set / control
```

Comment:
Maybe I missed it, but don't we call this anywhere?

Reply:
You missed it: https://github.com/input-output-hk/ouroboros-consensus/pull/140/files#diff-15b3ca76b62fd3e9d2f3c38db8e93d4be13f708a88dabbbdb6878092d026f0e8R121

and call it here: https://github.com/input-output-hk/ouroboros-consensus/pull/140/files#diff-f4876414f45096e37653d15a0c412c47a6e1c7fc7d62cb87560b3ccb3a03f2a5R1002